addons/danger/functions/fnc_doFlare.sqf

```diff
@@ -0,0 +1,98 @@
+#include "script_component.hpp"
+/*
+ * Author: nkenny
+ * Picks one unit from an array to shoot flare if available
+ *
+ * Arguments:
+ * 0: Units  <ARRAY>, <GROUP> or <OJECT>
+ *
+ * Return Value:
+ * available units
+ *
+ * Example:
+ * [units bob] call lambs_danger_fnc_doFlare;
+ *
+ * Public: No
+*/
+
+params ["_units"];
+
+// single unit
+if (_units isEqualType objNull) then {_units = [_units]};
+if (_units isEqualType grpNull) then {_units = units _units};
+
+// find grenade launcher
+private _flare = "";
+private _muzzle = "";
+private _unit = _units findIf {
+
+    private _weapon = primaryWeapon _x;
+
+    if !(_weapon isEqualTo "") then {
+
+        _muzzle = (getArray (configfile >> "CfgWeapons" >> _weapon >> "muzzles") - ["SAFE", "this"]) param [0, ""];
+
+        // find flares
+        if !(_muzzle isEqualTo "") then {
+            private _findFlares = getArray (configfile >> "CfgWeapons" >> _weapon >> _muzzle >> "magazines");
+            _findFlares = _findFlares arrayIntersect (magazines _x);
+            if (_findFlares isEqualTo []) exitWith {false};
+
+            // sort flares
+            private _index = _findFlares findIf {
+                private _ammo = getText (configfile >> "CfgMagazines" >> _x >> "Ammo");
+                private _flareSize = getNumber (configfile >> "CfgAmmo" >> _ammo >> "flareSize");
+                _flareSize != 0
+            };
+            
+            if (_index == -1) exitWith {false};
+            _flare = _findFlares select _index;
+        };
+    };
+    !(_flare isEqualTo "")
+};
+
+// execute
+if (_unit == -1) exitWith {_units};
+_unit = _units select _unit;
+
+// force
+doStop _unit;
+_unit setUnitPosWeak "MIDDLE";
+_unit setVariable [QGVAR(forceMove), true];
+
+// variable
+_unit setVariable [QGVAR(currentTask), "Shoot flare"];
+
+// dummy ~ seems necessary to get the AI to shoot up! -nkenny
+private _dummy = "Sign_Sphere10cm_F" createvehicle [0, 0, 0];
```

Comment:
this still should be a invisible object and not visible.
also it is createVehicle and setPos

Reply:
setPos is still Missing

Reply:
What do you mean? 

Also: I've experimented with various objects, including simpleObjects and so on. Pretty much the only way to consistently get the AI to target the object is to make a proper one. I opted not to use createVehicleLocal, as I understand that this is not open on all servers.  Hence, I think spawning in a 10cm sphere in yellow -- in a dark nightsky, is  a small price to pay. 

I guess we could give it transparent textures -- but why mess with the extra traffic. It should presumably be dark in either case (and 200 meters up!)

Reply:
you write setpos and it is setPos